api/v1alpha1/datadogagent_default.go

```diff
@@ -532,52 +533,51 @@ func DefaultDatadogAgentSpecAgentImage(image *ImageConfig) *ImageConfig {
 
 // DefaultDatadogAgentSpecAgentConfig used to default a NodeAgentConfig
 // return the defaulted NodeAgentConfig
-func DefaultDatadogAgentSpecAgentConfig(config *NodeAgentConfig) *NodeAgentConfig {
-	if config == nil {
-		config = &NodeAgentConfig{}
+func DefaultDatadogAgentSpecAgentConfig(agent *DatadogAgentSpecAgentSpec) *NodeAgentConfig {
+	if agent.Config.LogLevel == nil {
+		agent.Config.LogLevel = NewStringPointer(DefaultLogLevel)
 	}
 
-	if config.LogLevel == nil {
-		config.LogLevel = NewStringPointer(DefaultLogLevel)
+	if agent.Config.CollectEvents == nil {
+		agent.Config.CollectEvents = NewBoolPointer(defaultCollectEvents)
 	}
 
-	if config.CollectEvents == nil {
-		config.CollectEvents = NewBoolPointer(defaultCollectEvents)
+	if agent.Config.LeaderElection == nil {
+		agent.Config.LeaderElection = NewBoolPointer(defaultLeaderElection)
 	}
 
-	if config.LeaderElection == nil {
-		config.LeaderElection = NewBoolPointer(defaultLeaderElection)
+	if agent.Config.Resources == nil {
+		agent.Config.Resources = &corev1.ResourceRequirements{}
 	}
 
-	if config.Resources == nil {
-		config.Resources = &corev1.ResourceRequirements{}
+	if agent.Config.CriSocket == nil {
+		agent.Config.CriSocket = &CRISocketConfig{}
 	}
 
-	if config.CriSocket == nil {
-		config.CriSocket = &CRISocketConfig{
-			DockerSocketPath: NewStringPointer(defaultDockerSocketPath),
+	// Don't default Docker/CRI paths with Agent >= 7.27.0
+	// Let Env AD do the work for us
+	agentTag := utils.GetTagFromImageName(agent.Image.Name)
+	if !(agentTag == "latest" || utils.IsAboveMinVersion(agentTag, "7.27.0")) {
```

Comment:
does this support agent 6 and jmx images?

Reply:
Good catch, changed.